engine/src/main/java/org/destinationsol/game/ship/hulls/Hull.java

```diff
@@ -158,22 +159,42 @@ public void onRemove(SolGame game) {
         if (myEngine != null) {
             myEngine.onRemove(game, myPos);
         }
+        myParticleEmitters.forEach(pe -> pe.onRemove(game, myPos));
 
     }
 
-    public void setEngine(SolGame game, SolShip ship, Engine ei) {
+    public void setEngine(SolGame game, SolShip ship, Engine engine) {
         List<Drawable> drawables = ship.getDrawables();
         if (myEngine != null) {
```

Comment:
Add a comment above the if: `Remove the old engine and its associated drawables`